composeApp/src/desktopMain/kotlin/org/ooni/probe/Main.kt

```diff
@@ -12,20 +13,28 @@ import io.github.vinceglb.autolaunch.AutoLaunch
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 import ooniprobe.composeapp.generated.resources.Res
 import ooniprobe.composeapp.generated.resources.app_name
 import ooniprobe.composeapp.generated.resources.tray_icon
 import org.jetbrains.compose.resources.painterResource
 import org.jetbrains.compose.resources.stringResource
+import org.ooni.probe.data.models.DeepLink
+import java.awt.Desktop
 
 fun main() {
     application {
         val autoLaunch = AutoLaunch(appPackageName = "org.openobservatory.ooniprobe")
 
         var isWindowVisible by remember { mutableStateOf(!autoLaunch.isStartedViaAutostart()) }
+        val deepLinkFlow = MutableSharedFlow<DeepLink?>(extraBufferCapacity = 1)
+        val deepLink by deepLinkFlow.collectAsState(null)
 
+        Desktop.getDesktop().setOpenURIHandler { event ->
```

Comment:
`val autoLaunch`, `val deepLinkFlow` and `setOpenURIHandler` should be called outside of `application { }`. Because inside they will run again if there's any re-composition.